app/src/main/java/com/example/plantsapp/presentation/ui/profile/ProfileFragment.kt

```diff
@@ -51,8 +56,26 @@ class ProfileFragment : Fragment(R.layout.fragment_profile) {
             )
 
             tvProfileName.text = user.name
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_profile_appbar, menu)
+    }
 
-            btnSignOut.isEnabled = true
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_sign_out -> {
+                viewModel.onSignOutClick()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
         }
     }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        menu.findItem(R.id.action_sign_out)
+            .isVisible = isSignOutBtnVisible
```

Comment:
Maybe try to observe viewModel isSignOutBtnVisible state from viewmodel instead of storing it in fragment as variable

You can try to observe in onCreateOptionsMenu